osquery/tables/events/linux/bpf_file_events.cpp

```diff
@@ -0,0 +1,179 @@
+                for (auto file : files) {
+                    replaceGlobWildcards(file);
+                    //std::cout << "File pattern: " << file << std::endl;
+
+                    std::vector<std::string> file_path_list = {};
+                    resolveFilePattern(file, file_path_list);
+
+                    //for (const auto& file_path : file_path_list) {
+                    //    std::cout << "\t File path: " << file_path << std::endl;
+                    //}
+
+                    monitored_paths.insert(monitored_paths.end(), file_path_list.begin(), file_path_list.end());
+                }
+    });
+}
+
+void BPFFileEventSubscriber::configure() {
+    configure_filtering();
+}
+
+Status BPFFileEventSubscriber::init() {
+  auto subscription_context = createSubscriptionContext();
+  subscribe(&BPFFileEventSubscriber::eventCallback, subscription_context);
+
+  return Status::success();
+}
+
+Status BPFFileEventSubscriber::eventCallback(const ECRef& event_context,
+                                               const SCRef&) {
+  auto row_list = generateRowList(event_context->event_list);
+  addBatch(row_list);
+
+  return Status::success();
+}
+
+bool BPFFileEventSubscriber::generateRow(
+    Row& row, const ISystemStateTracker::Event& event) {
+  row = {};
+
+  switch (event.type) {
+  case ISystemStateTracker::Event::Type::Open:
+    row["syscall"] = TEXT("open");
+    break;
+
+  default:
+    return false;
+  }
+
+  bool is_path_matched = false;
+
+  if (!std::holds_alternative<ISystemStateTracker::Event::FileIOData>(event.data)) {
+    row["file_path"] = TEXT("");
+    row["flags"] = TEXT("");
+
+    } else {
+      const auto& file_io_data =
+        std::get<ISystemStateTracker::Event::FileIOData>(event.data);
+
+      row["file_path"] = TEXT(file_io_data.file_path);
+      row["flags"] = TEXT(file_io_data.flags);
+
+      if (file_io_data.flags.find("O_CREAT") != std::string::npos)
+      {
+          configure_filtering();
+      }
+
+      for (const auto& match_path : monitored_paths)
+      {
+          if (boost::algorithm::starts_with(file_io_data.file_path, match_path))
+          {
+              is_path_matched = true;
+              break;
+          }
+      }
+    }
+
+  if (!is_path_matched)
+      return false;
+
+  row["ntime"] = TEXT(event.bpf_header.timestamp);
+  row["tid"] = INTEGER(event.bpf_header.thread_id);
+  row["pid"] = INTEGER(event.bpf_header.process_id);
+  row["uid"] = INTEGER(event.bpf_header.user_id);
+  row["gid"] = INTEGER(event.bpf_header.group_id);
+  row["cid"] = INTEGER(event.bpf_header.cgroup_id);
+  row["fd"] = INTEGER(event.bpf_header.exit_code);
+  row["probe_error"] = INTEGER(event.bpf_header.probe_error);
+  row["parent"] = INTEGER(event.parent_process_id);
+  row["path"] = TEXT(event.binary_path);
+  row["duration"] = INTEGER(event.bpf_header.duration);
+  row["uptime"] = TEXT(std::to_string(getUptime()));
+
+  if (!std::holds_alternative<ISystemStateTracker::Event::FileIOData>(
```

Comment:
This was already evaluated earlier, only check it once. The entire block can be removed